### ISSUE TITLE
Allow selecting element by clicking on the outer border

### ISSUE THREAD

**Issue by miina** [GoogleForCreators/web-stories-wp#5325]
## Feature Description
Initially, the outer border was implemented so that it wouldn't be possible to select an element clicking on it, however, after consulting with Sam, the expected behavior is to allow selecting an element by clicking on the border as well.

Additionally, in discussion with George, we should include the border to the selection line as well.

- [ ] Include the border into selection and allow clicking on the border to select it.

## Additional Context

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance Criteria
- The user should be able to select element by clicking on the outer border.

## Implementation Brief

The element needs to look exactly the same in the editor and in the front-end, the only thing that changes visually is the selection line including the border now, and the user being able to select the element by clicking on the border.

This can be achieved in a similar way as we're adding a padding:
- When border is added, the width and height of the element are increased accordingly, and decreased when the border is removed.
- The position is also adjusted accordingly, also considering rotation

We should additionally add some padding to the element to allow the size of the element content to stay the same.

The text size calculation needs to consider the text element content's size only (see `calculateTextHeight` and `calculateFitTextFontSize`)

To investigate: Edit modes for media and text -- probably should be as currently since only the content can be manipulated.

To investigate: option of adjusting the frame only? Probably better to keep frame and display in sync due to snapping, dragging, and generally not to complicate the readability.

To investigate: Still need to investigate how the shapes work since those have different implementation. It's possible that non-rectangular shapes will stay as they are -- the selection line is currently in the middle anyway so it's partially clickable and different from rectangular elements already now.

**Comment by miina** [comment]
@barklund FYI added this issue after discussion with Sam.

**Comment by miina** [comment]
Since #6171 is now being worked on, will mark this blocked by that issue as we should ensure the two will work together ok. Moving it out of the sprint for now as #6171 was pulled in again.

**Comment by miina** [comment]
@merapi I've updated the implementation brief for a general overview.
Two points:
1. Edit mode for media: it seems to me that it might make sense to leave the media edit mode as it is since cropping and panning is for the media part only anyway so might be good to leave the border out of it, especially if it should be easier technically. WDYT?
2. Non-rectangular shapes: these currently already have the selection line somewhere in the middle of the border so if it gets too complicated to implement it for those then I'd suggest splitting the non-rectangular shapes into a separate issue. It's anyway already in the middle so it differs from other elements already now and is already now partially clickable, so merging the rectangular elements separate should be ok. Thoughts?

**Comment by merapi** [comment]
@miina 
1. I agree - leave the edit mode as is.
2. I'm not sure if this border in the middle was this intentional or just it just happened, with very big borders you kinda lose the shape. Feel free to split it if needed.
